crates/walrus-service/src/client/cli/args.rs

```diff
@@ -1011,6 +1138,139 @@ impl std::fmt::Display for BlobIdentity {
     }
 }
 
+/// Query for quilt patches by identifier.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByIdentifier {
+    /// The quilt ID, which is the BlobID of the quilt.
+    pub quilt_id: BlobId,
+    /// The identifiers to read from the quilt.
+    pub identifiers: Vec<String>,
+}
+
+/// Query for quilt patches by tag.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByTag {
+    /// The quilt ID, which is the BlobID of the quilt.
+    pub quilt_id: BlobId,
+    /// The tag key.
+    pub tag: String,
+    /// The tag value.
+    pub value: String,
+}
+
+/// Query for quilt patches by patch ID.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByPatchId {
+    /// The quilt patch IDs.
+    pub quilt_patch_ids: Vec<QuiltPatchId>,
+}
+
+/// Selector for quilt patches.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum QuiltPatchSelector {
+    /// Patches by quilt_id and identifier.
+    ByIdentifier(QuiltPatchByIdentifier),
+    /// Patches by quilt_id and tag.
+    ByTag(QuiltPatchByTag),
+    /// Patches by quilt_patch_id.
+    ByPatchId(QuiltPatchByPatchId),
+}
+
+/// Query parameters to read quilt patches.
+#[serde_as]
+#[derive(Debug, Clone, Args, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchQuery {
```

Comment:
Question/suggestion: Could we use a similar construction as for `NodeSelection` with `#[group(required = true, multiple = false)]`? Alternatively, you could add some `conflicts_with` attributes to ensure that some of the options cannot be specified simultaneously.

Also, it would be good to allow just specifying the quilt ID (without identifiers or tags, in which case all files would be read).